### DISCUSSION TITLE
how to speed up `gino.create_engine`

### DISCUSSION THREAD

**Discussion by panla** [python-gino/gino#774] [Q&A]
### costs too much time

```python
import gino

engine = await gino.create_engine(...)
```

if I want to change engine when i write data, `gino.create_engine` costs too much time .
is there a better idea to speed up ?

**Comment by mikekeda** [comment]
you could keep it open and reuse, so you don't need to open it for each request

**Reply by panla** [reply]
thanks, but I want to select data by a engine and `create/update/delete` data by another engine.

**Reply by wwwjfy** [reply]
engine only needs to be initialized once and kept references

**Reply by panla** [reply]
> engine only needs to be initialized once and kept references

```
import gino
from gino_starlette import Gino

db = Gino(...)

async def get_write_engine():
    return await gino.create_engine(WRITE_DSN)

engine = None

async def save(self):

    if WRITE_DSN != READ_DSN:
        global engine
        bind = engine or await get_write_engine()
        engine = bind
        obj = await self.create(bind=engine)
        return obj
    return await self.create()
```

effective
